charts/dawarich/templates/photon-loadbalancer/deployment.yaml

```diff
@@ -0,0 +1,77 @@
+{{- if .Values.photon.enabled }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "common.fullname" . }}-photon-loadbalancer
+  labels:
+    app.kubernetes.io/component: "photon-loadbalancer"
+    {{- include "common.labels" . | nindent 4 }}
+spec:
+  revisionHistoryLimit: 0
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/component: "photon-loadbalancer"
+      {{- include "common.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/component: "photon-loadbalancer"
+        {{- include "common.selectorLabels" . | nindent 8 }}
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/photon-loadbalancer/configmap.yaml") . | sha256sum }}
+    spec:
+      securityContext: {{ toYaml .Values.photon.loadbalancerProxy.podSecurityContext | nindent 8 }}
+      containers:
+      - name: envoy
+        image: '{{ .Values.photon.loadbalancerProxy.image.registry }}/{{ .Values.photon.loadbalancerProxy.image.repository }}:{{ .Values.photon.loadbalancerProxy.image.tag }}'
+        imagePullPolicy: {{ .Values.photon.loadbalancerProxy.image.pullPolicy }}
+        command:
+        - /usr/local/bin/envoy
+        args:
+        - --config-path
+        - /etc/envoy/envoy.yaml
+        - --service-cluster
+        - photon-proxy
```

Comment:
The naming is inconsistent, it's now sometimes called photon-loadbalancer and sometimes photon-proxy. Let's unify on photon-gateway as the final term. In values.yaml this can be shortened to gateway so values would be photon.gateway.image for example.